### PR TITLE
Updates for Pottercraft and fixes to Nox

### PR DESCRIPTION
**Pull request by Azami7**
- fixed knut material
- added new event fired when players try a wand that isnt their destined wand
- changed player wand wood/core accessors so they dont return null
- made wandLoreConjunction public for use elsewhere
- added a get random type for O2WandCoreType and O2WandWoodType
- changed OllivandersAPI to only make objects if they are not initialized... derp
- made Nox also target the caster
- updated description on RemovePotionEffect
- fixed targeted self so it also still targets others for RemovePotionEffectInRadius

### CODE REVIEW

File: Ollivanders/src/net/pottercraft/ollivanders2/spell/RemovePotionEffectInRadius.java [hunk 1]
```diff
@@ -57,10 +57,8 @@ public void checkEffect()
         {
             removePotionEffects(player);
         }
-        else
```
Comment (autumnwoz) [inline Ollivanders/src/net/pottercraft/ollivanders2/spell/RemovePotionEffectInRadius.java:60]: Why did we remove the else?

Comment (Azami7) [reply Ollivanders/src/net/pottercraft/ollivanders2/spell/RemovePotionEffectInRadius.java:60]: we want it to do both - affect the person and also affect the radius, not one or the other

Comment (Azami7) [reply Ollivanders/src/net/pottercraft/ollivanders2/spell/RemovePotionEffectInRadius.java:60]: when affect self is toggled - some we only want to affect everyone else, this is why the caster is handled specifically

Comment (autumnwoz) [reply Ollivanders/src/net/pottercraft/ollivanders2/spell/RemovePotionEffectInRadius.java:60]: There are no cases where we only want it to affect the player?

Comment (Azami7) [reply Ollivanders/src/net/pottercraft/ollivanders2/spell/RemovePotionEffectInRadius.java:60]: You mean only the player - not yet though if there is I will add an affectOnlyPlayer and then put a conditional around this.

**Pull request merged**